### ISSUE TITLE
Change terminology from "continuable failure" to "soft failure"

### ISSUE THREAD

**Issue by pekkaklarck** [robotframework/robotframework#3692]
Robot Framework has for long time had support for failures that allow execution it to continue but mark the test failed at the end. This functionality is supported also by many other tools such as [TestNG](https://www.javadoc.io/doc/org.testng/testng/6.8.8/org/testng/asserts/SoftAssert.html) and PyTest via [various](https://pypi.org/project/softest/) [plugins](https://github.com/okken/pytest-check). For general benefits of this functionality see https://assertless.org/.

We have used our own term "continuable failure" with these failures but a more commonly (albeit not uniformly) used term in our domain is "soft assert". To make this functionality easier to discover, we should use the commonly used term as well. We don't generally use word "assert" as it's more a unit testing term, but using "soft failure" would work very well in our lingo.

What I propose is the following:

- Update User Guide section about [continuable failures](http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#continue-on-failure) to use the term "soft failure". The current section should probably be split to separate sections about ignoring failures and soft failures.

- Leave the existing keyword `Run Keyword And Continue On Failure` as-is as works just fine even if the overall term for the functionality would be soft failure. This term should be mentioned in the keyword documentation, though.

- Leave the special `ROBOT_CONTINUE_ON_FAILURE` attribute custom exceptions can use to mark they are continuable/soft as-is. Also it works fine even after the terminology change and changing it would require a deprecation period.

- Name the planned new special exception activating this functionality `SoftFailure`. See #3685 for more details about these plans.

In other words, there shouldn't be need for any functional change to existing code.

**Comment by pekkaklarck** [comment]
A small problem is that the term "soft failure" or "soft error" is used with electronics (e.g. RAM) when there's a failure caused by cosmic rays or other such particles:
https://en.wikipedia.org/wiki/Soft_error

I think this is separate enough domain that we using "soft failure" to mean something different is fine.

**Comment by humbienri** [comment]
How about "NonTerminating Failure" or something to that effect. I think both Soft Failure and Continuable Failure would suffer from the same issue of having to explain what that means, particularly to businessy folks such as CTOs or Product Owners. I'm not sure that you gain very much by going to Soft Failure to merit the change but others can surely feel differently.

**Comment by pekkaklarck** [comment]
Decided not to do this change for the following reasons:

- There hasn't been that much interest towards the change.
- "Continuable falure" may not be the best term but it nevertheless explains the concept more explicitly than "soft failure".
- The change would require some extra work and could cause confusion.

This can be considered again in the future if "soft failure" or some other term becomes more widely used. Switching to some other new term like "non-terminating failure" would likely not be worth the effort compared to the current situation.
